archive/extract_test.go

```diff
@@ -12,49 +12,29 @@ import (
 
 	"github.com/sclevine/spec"
 	"github.com/sclevine/spec/report"
+	"golang.org/x/sync/errgroup"
 
 	"github.com/buildpacks/lifecycle/archive"
 	h "github.com/buildpacks/lifecycle/testhelpers"
 )
 
+var originalUmask int
+
 func TestArchiveExtract(t *testing.T) {
 	rand.Seed(time.Now().UTC().UnixNano())
+	originalUmask = h.GetUmask(t)
```

Comment:
I personally like this style for tests. 

```
originalUmask = setUmask(0)
_ = setUmask(originalUmask)
```

Reply:
Yeah, we thought about it. @natalieparellano, it's your call. I really don't mind. Maybe what @jabrown85 suggested is clearer.

Reply:
That works for me!

Reply:
This would allow us to delete our `GetUmask()` helpers.

Reply:
Thinking about it again, I'm not sure if it's possible to get rid of the `GetUmask` functions because of [this line](https://github.com/buildpacks/lifecycle/pull/730/files#diff-b12673f7f2d9439d536db38a5a66d22c553089b91aaa34501d19f34e4743edb3R94).
If we'll try to replace that line with:
```
originalUmask = setUmask(0)
_ = setUmask(originalUmask)
```
Then we might have problems with other tests that run in parallel.

And if we don't get rid of this function, I don't think it's worth changing [the suggested line](https://github.com/buildpacks/lifecycle/pull/730/files#diff-b12673f7f2d9439d536db38a5a66d22c553089b91aaa34501d19f34e4743edb3R25).

@jabrown85 @natalieparellano, what do you think?

Reply:
Ah, you're right!

Reply:
I think we should just leave it as is unless any of you (or someone else) has a better solution.

Reply:
After talking with @jabrown85, we decided to keep the code as is.